### DISCUSSION TITLE
Build notifications not sent to BitBucket for manually triggered builds

### DISCUSSION THREAD

**Discussion by banjoh** [jenkinsci/bitbucket-push-and-pull-request-plugin#228] [Ideas]
At times builds fail due to some flaky environment problem which is inevitable. Such builds may end up being marked as failed or unstable. To resolve this, one might restart the build manually to get the build to pass. Unfortunately the build status does not get updated.

According to https://support.atlassian.com/bitbucket-cloud/docs/integrate-your-build-system-with-bitbucket-cloud/, it does not look like there are any specific environment variables that could be missing for manually triggered builds.

Is there a way to have the plugin handle manually triggered builds for the purpose of build notifications to BB? Currently we have a separate script that does this for us.

### Environment
Jenkins: 2.235.1
BPPP: 2.4.0

PS: I can submit a patch if need be, but I'd need some pointers of what needs to be done

**Comment by Gozke** [comment]
I'd also find this feature useful.
Together with this suggested feature I think it would be very useful if it was possible to use the job's name as _key_ value in the status update request payload as opposed to the build number. Here's why:
We have a couple of repositories that are all integrated with Jenkins the same way:

- Bitbucket triggers a jenkins job upon _repo:refs_changed_, _pr:opened_ and _pr:from_ref_updated_
- The jenkins job builds the repo content, runs unit test AND performs Sonar analysis
- Build status is back-propagated to Bitbucket.
- In Bitbucket pull requests can only be merged if
   - All reviewers approved it
   - There are no running builds (for any of the involved commits)
   - There's at least one build for the most recent commit with SUCCESS state and there's no failed build for this commmit

Now, sometimes the triggered build fails. E.g.: a unit test that has some time constraint runs too long, because of high load on the build server, Sonar is not available, etc.. In such cases if I manually start the job I'd like it to update the previous build's status and not add a new successful build. (By the way this is how the _atlassian-bitbucket-server-integration-plugin_ works too)
I could imagine a global configuration flag with a name like 'use job name as build key' or something. Does this make sense to you all?

**Comment by y0zg** [comment]
This feature is really needed to make this plugin mature
